src/command.ts

```diff
@@ -59,49 +71,48 @@ export type ComparisonResult = {
   percentage: number
 }
 
-export type UpdateBaseResult = {
-  baseUpdated: boolean
-}
-
 /** Call the plugin to compare snapshot images and generate a diff */
 function compareScreenshots(name: string, screenshotOptions: any): Chainable<ComparisonResult> {
-  const errorThreshold = getErrorThreshold(screenshotOptions)
-  const options: CompareSnapshotsPluginArgs = {
-    fileName: name,
-    errorThreshold,
-    // @ts-expect-error TODO fix potential null error
-    specRelativePath: Cypress.config().spec.relative,
-    specFolder: Cypress.env('visualRegression').specFolder,
-    baseDirectory: Cypress.env('visualRegression').baseDirectory,
-    diffDirectory: Cypress.env('visualRegression').diffDirectory,
-    generateDiff: Cypress.env('visualRegression').generateDiff,
-    failSilently: false
-  }
-
-  if (screenshotOptions.failSilently !== null) {
-    options.failSilently = screenshotOptions.failSilently
-  } else if (Cypress.env('failSilently') !== null) {
-    options.failSilently = Cypress.env('failSilently')
-  }
+  return cy.get('@screenshotAbsolutePath').then((screenshotAbsolutePath: unknown) => {
+    if (typeof screenshotAbsolutePath !== 'string') {
+      throw new Error('Could not resolve screenshot path')
+    }
+    const errorThreshold = getErrorThreshold(screenshotOptions)
+    const options: CompareSnapshotsPluginArgs = {
+      screenshotName: name,
+      errorThreshold,
+      // @ts-expect-error TODO fix potential null error
+      specName: Cypress.config().spec.name,
+      screenshotAbsolutePath,
+      baseDirectory: Cypress.env('visualRegression').baseDirectory,
+      diffDirectory: Cypress.env('visualRegression').diffDirectory,
+      generateDiff: Cypress.env('visualRegression').generateDiff
+    }
 
-  // eslint-disable-next-line promise/catch-or-return
-  return cy.task('compareSnapshotsPlugin', options).then((results: any) => {
-    if (results.error !== undefined && options.failSilently === false) {
-      throw deserializeError(results.error)
+    let failSilently = false
+    if (screenshotOptions.failSilently !== undefined) {
+      failSilently = screenshotOptions.failSilently
+    } else if (Cypress.env('visualRegression').failSilently !== undefined) {
+      failSilently = Cypress.env('visualRegression').failSilently
     }
-    return results
+
+    return cy.task('compareSnapshotsPlugin', options).then((results: any) => {
+      if (results.error !== undefined && !failSilently) {
+        throw deserializeError(results.error)
```

Comment:
One of the goals I also have is to push all the control of different flows to the command.ts level, rather then the plugin.ts level, as if there's no explicit need to delegate something to plugin.ts (wihin node process) we shouldn't do it 🤷